server/routes/dsl_route.ts

```diff
@@ -100,10 +106,17 @@ export function registerDslRoute(router: IRouter) {
 
         const start = performance.now();
         try {
-          const resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
-            'search',
-            params
-          );
+          let resp;
+          if(dataSourceEnabled && dataSourceId2){
```

Comment:
looks like these two queries are async, is it whichever query return first then it will show on the dashboard first?